packages/system/src/Item/Components/ItemSearchInput/ItemSearchInput.tsx

```diff
@@ -4,9 +4,33 @@ import {
   useFormatNumber,
   useTranslation,
   styled,
+  Autocomplete,
```

Comment:
This got a little hairy!

The `ItemSearchInput` was expecting an `Item` to be passed as the 'current item'. The stocktake is now using an `ItemRowFragment`!

So.. went back to the `toItem` casting .. hack? which just creates the `Item` type to make typescript happy. In general I think this component is confusing and over complicated for what it needs to do.

Really, just passing an `id` for the `item` and doing a `.find` in the items returned would be a bit easier

Reply:
what about going the other way, and changing the list query to return `ItemRowFragment[]`?
Would that help? It would mean adding `availableQuantity` to the fragment.. the only props used are `code` `name` and `availableQuantity` in here.. and I think consumers of the component would only need the `id`. 

Just a thought - the current `toItem` is all good!

Reply:
Yeah that's far better. I've had a quick look at changing that around and it will require changes in multiple places which are using the search input and a pretty big diff. 

So..  new issue to revisit when things are a little more stable.. 🙄